python/idsse_common/test/test_grid_proj.py

```diff
@@ -0,0 +1,146 @@
+"""Test suite for grid_proj.py"""
+# --------------------------------------------------------------------------------
+# Created on Wed Aug 2 2023
+#
+# Copyright (c) 2023 Colorado State University. All rights reserved. (1)
+#
+# Contributors:
+#     Mackenzie Grimes (1)
+#
+# --------------------------------------------------------------------------------
+# pylint: disable=missing-function-docstring,redefined-outer-name,invalid-name,protected-access
+
+from typing import Tuple
+
+from pytest import fixture, approx
+
+from idsse.common.grid_proj import GridProj, PixelRounding
+
+# example data
+EXAMPLE_PROJ_SPEC = '+proj=lcc +lat_0=25.0 +lon_0=-95.0 +lat_1=25.0 +r=6371200'
+PROJ_SPEC_WITH_OFFSET = (
+    '+proj=lcc +lat_0=25.0 +lon_0=-95.0 +lat_1=25.0 +x_0=3271.152832031251 '
+    '+y_0=263.7934687500001 +r=6371.2, +units=km'
+)
+EXAMPLE_GRID_SPEC = "+dx=2.539703 +dy=2.539703 +w=2345 +h=1597 +lat_ll=19.229 +lon_ll=-126.2766"
+
+EXAMPLE_PIXELS = [
+    (0, 0),
+    (0, 1),
+    (2000, 1500)
+]
+EXAMPLE_PROJECTIONS = [
+    (-126.2766, 19.229),
+    (-126.27660549554, 19.229022224607),
+    (-126.23803580508, 19.272773488997)
+]
+EXAMPLE_GEOS = [
+    (-3275807.350733, -260554.63043505),
+    (-3275807.350733, -260552.09073205),
+    (-3270727.944733, -256745.07593505)
+]
+
+
+# utility to roughly compare tuples of floats with less floating point precision
+def approx_tuple(values: Tuple[float, float]) -> Tuple:
+    return (approx(values[0]), approx(values[1]))
+
+
+# fixtures
+@fixture
+def grid_proj() -> GridProj:
+    return GridProj.from_proj_grid_spec(EXAMPLE_PROJ_SPEC, EXAMPLE_GRID_SPEC)
+
+
+# test class methods
+def test_from_proj_grid_spec(grid_proj: GridProj):
+    assert isinstance(grid_proj, GridProj)
+
+    assert (grid_proj._x_offset, grid_proj._y_offset) == approx_tuple(EXAMPLE_GEOS[0])
+    assert grid_proj._h == 1597
+    assert grid_proj._w == 2345
+    assert grid_proj._dx == approx(2.539703)
+    assert grid_proj._dy == grid_proj._dx
+
+    t = grid_proj._trans
+    assert t.source_crs is not None and t.source_crs.type_name == 'Geographic 2D CRS'
+    assert t.target_crs is not None and t.target_crs.type_name == 'Projected CRS'
+
+
+def test_from_proj_grid_spec_with_offset():
+    proj_with_offset = GridProj.from_proj_grid_spec(PROJ_SPEC_WITH_OFFSET, EXAMPLE_GRID_SPEC)
+    assert proj_with_offset._x_offset == approx(-3272536.1979)
+    assert proj_with_offset._y_offset == approx(-260290.8369)
+
+
+# transformation methods testing
+def test_map_proj_to_pixel(grid_proj: GridProj):
```

Comment:
Should include testing rounding, once you add rounding option to map_proj_to_pixel().

Reply:
This was still using the incorrect round function, good catch. Fixed now.